core/src/main/java/software/amazon/awssdk/RequestConfig.java

```diff
@@ -58,11 +60,11 @@
     public abstract String getRequestType();
 
     /**
-     * @return The original request object. May be delivered to various strategies or hooks for
-     *     extra context. I.E. {@link RequestHandler} or {@link
-     * RetryPolicy}.
+     * @return The original request object, before any modifications by {@link ExecutionInterceptor}s. This should only really
+     * be used for replaying requests in the case of retries. All other cases should use the request as it was modified by the
```

Comment:
This comments seems misleading. We replay the marshalled request on retries, we don't marshall again.